src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -228,6 +241,51 @@ def get_update_payment_status_data(
         return render_to_json(options["payment_status_update_json"], context)
 
 
+class ObjectsAPIV2Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV2]):
+
+    @override
+    def get_object_data(
+        self, submission: Submission, options: RegistrationOptionsV2
+    ) -> dict[str, Any]:
+        submission_variables = get_variables_for_context(submission)
+        variables_mapping = options["variables_mapping"]
+        record_data: dict[str, Any] = {}
+
+        for mapping in variables_mapping:
+            variable_key = mapping["variable_key"]
+            target_path = mapping["target_path"]
+
+            # Type hint is wrong: currently some static variables are of type date/datetime
+            value = cast(Any, submission_variables[variable_key])
+
+            # Comply with JSON Schema "format" specs:
+            if isinstance(value, (datetime, date)):
+                value = value.isoformat()
```

Comment:
This is a bit surprising: some static variables return `date` or `datetime` objects, while all the type annotations related to this were saying it was JSON data (e.g. here `submission_variables` is typed as `JSONValue`). I think this relates again to proper type checking of variables/JSON Logic

Reply:
That's why we created #2324 :(